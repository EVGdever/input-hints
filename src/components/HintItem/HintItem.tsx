import React, {Dispatch, SetStateAction, useEffect, useState} from "react";
import {UserInterface} from "../../App";

export interface HintItemProps {
  index: number,
  active: boolean,
  user: UserInterface,
  clickHandler(id: number): void,
  setActiveHint: Dispatch<SetStateAction<number>>,
}

const HintItem: React.FC<HintItemProps> = ({
  user,
  active,
  clickHandler,
  setActiveHint,
  index
}) => {
  const [img, setImg] = useState(undefined);

  useEffect(() => {
    // Get user image
    fetch('https://jsonplaceholder.typicode.com/photos/' + user.id)
      .then(response => response.json())
      .then(json => setImg(json.url))
  }, [user])

  const hoverHandler = (id: number) => {
    setActiveHint(id)
  }

  return (
    <div
      className={"hint-item" + (active ? ' active' : '')}
      onClick={() => clickHandler(index)}
      onMouseEnter={() => hoverHandler(index)}
    >
      {
        img ?
          <>
            <img className="avatar" src={img} alt=""/>
            <div className="user-info">
              <p className="name">{user.name}</p>
              <p className="email">{user.email}</p>
            </div>
          </> : null
      }
    </div>
  )
}

export default HintItem;