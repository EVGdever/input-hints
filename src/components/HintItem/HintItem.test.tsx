import React from 'react';
import {shallow, mount} from 'enzyme';
import HintItem from "./HintItem";

let activeHint = 0

const props = {
  user: {
    email: "Sincere@april.biz",
    id: '1',
    name: "Leanne Graham"
  },
  activeHint,
  clickHandler: (id: number) => {
    console.log(id)
  },
  index: 0,
  setActiveHint() {
    activeHint++;
  }
}

describe('HintItem', () => {
  it('should render a hint-item', () => {
    const component = shallow(
      <HintItem
        key={props.user.id}
        user={props.user}
        index={props.index}
        clickHandler={props.clickHandler}
        setActiveHint={props.setActiveHint}
        active={props.activeHint === props.index}
      />
    )

    expect(component).toMatchSnapshot();
  });
})