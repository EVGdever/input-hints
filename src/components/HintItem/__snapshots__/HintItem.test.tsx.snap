// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintItem should render a hint-item 1`] = `
<div
  className="hint-item active"
  onClick={[Function]}
  onMouseEnter={[Function]}
/>
`;
